lib/src/local_working_copy.rs

```diff
@@ -1591,7 +1603,19 @@ impl FileSnapshotter<'_> {
                 err: err.into(),
             })?;
             self.target_eol_strategy
-                .convert_eol_for_snapshot(BlockingAsyncReader::new(file))
+                .convert_eol_for_snapshot(BlockingAsyncReader::new(file), async || {
+                    old_file_ids
+                        .try_map_async(async |file_id| {
+                            let Some(file_id) = file_id else {
+                                return Ok(None);
+                            };
+                            self.store()
+                                .read_file(repo_path, file_id)
+                                .await
+                                .map(Option::Some)
+                        })
+                        .await
```

Comment:
Perhaps, we should also update the materialization function to test conflict contents one by one, instead of converting a materialized conflict data to CRLF.

btw, it seems that we don't apply EOL conversion to the conflict content written by `write_conflict()`. That's probably a bug.